File: ALGORITHMS/JAVA/Searching/Binary_Search.java
public class Binary_Search
{
  //Define a function for binary search
  
  public static int binarySearch(int arr[], int target, int si, int ei)
  {
    int ans = 0;
    
    //Base Case For Recursion
    
    if(si > ei) return -1;
    
    int mid = si + (ei-si)/2;
    
    //Comparing target with element present at index mid
    
    if(arr[mid] == target)
    {
      return mid;
    }
    
    //Checking position of target with respect to mid in order to call binarySearch fun. recursively.
    
    else if(arr[mid] > target)
    {
      ans = binarySearch(arr, target, si, mid-1);
    }
    
    else if(arr[mid] < target)
    {
      ans = binarySearch(arr, target, mid+1, ei);
    }  
    
    return ans;
    
  }
  public static void main(String args[])
  {
    int arr[] = {2,7,10,12,24,46,55,78};
    
    int ans  = binarySearch(arr, 7, 0, arr.length-1);
    
    if(ans == -1) System.out.println("The Target Is Not Found.");
    
    else System.out.println("The Target Is Found At Index : " + ans);
  }  
}  

File: ALGORITHMS/C language/Searching/binary_search.c
// C program to implement recursive Binary Search
#include <stdio.h>

// A recursive binary search function.

int binarySearch(int arr[], int si, int ei, int target)
{
	if (ei >= si) {
		int mid = si + (ei - si) / 2;

		// If the element is present at the middle
		// itself
		if (arr[mid] == target)
			return mid;

		// Element can only be present in the left subarray 
        // if it is smaller than the midpoint.
		if (arr[mid] > target)
			return binarySearch(arr, si, mid - 1, target);

		// Otherwise, the element could only 
        // be found in the right subarray.
		return binarySearch(arr, mid + 1, ei, target);
	}

	// We reach here when element is not
	// present in array
	return -1;
}

int main(void)
{
	int arr[] = {5,8,3,6,0,10,11};
	int n = sizeof(arr) / sizeof(arr[0]);
	int target = 1;
	int result = binarySearch(arr, 0, n - 1, target);
    // ternary operator
	(result == -1)
		? printf("Element is not present in array")
		: printf("Element is present at index %d", result);
	return 0;
}
